compiler/rustc_hir_analysis/src/coherence/builtin.rs

```diff
@@ -66,10 +76,9 @@ impl<'tcx> Checker<'tcx> {
 
 fn visit_implementation_of_drop(checker: &Checker<'_>) -> Result<(), ErrorGuaranteed> {
     let tcx = checker.tcx;
-    let header = checker.impl_header;
```

Comment:
can this be `let impl_trait_ref = checker.impl_header.trait_ref.instantiate_identity();` instead? To avoid having redundant information in the `Checker` struct?